modules/layers/src/path-layer/path-layer.js

```diff
@@ -145,13 +145,18 @@ export default class PathLayer extends Layer {
         data: props.data,
         getGeometry: props.getPath,
         positionFormat: props.positionFormat,
-        fp64: this.use64bitPositions()
+        fp64: this.use64bitPositions(),
+        dataChanged: changeFlags.dataChanged
       });
       this.setState({
         numInstances: pathTesselator.instanceCount,
         bufferLayout: pathTesselator.bufferLayout
       });
-      attributeManager.invalidateAll();
+      if (!changeFlags.dataChanged) {
```

Comment:
I am assuming this `if` should test if any thing else changed other than `changeFlags.dataChanged`, what if `changeFlags.dataChanged` is true and also some other flag, like fp64 too? Shouldn't we exclusively check for all other cases?

Reply:
If `changeFlags.dataChanged` is `true`, the base layer `updateState` calls `attributeManager.invalidateAll()`. If `changeFlags.dataChanged` is a range, then the user has specifically requested a partial update, and therefore responsible for making sure that the rest of the data are not changed (see remarks in documentation).

Reply:
👍